statviz/src/utils/table.ts

```diff
@@ -69,5 +78,83 @@ export function table<Row extends object>(f: Array<Row>) {
 
       return table(joinedTable);
     },
+    leftJoin: (foreignTable, column: keyof Row, foreignColumn: string) => {
+      const joinedTable = data.map((row) => {
+        const foreignIndex = foreignTable.data.findIndex(
+          (e) => e[foreignColumn] == row[column]
+        );
+        if (foreignIndex !== -1) {
+          return {
+            ...row,
+            ...foreignTable.data[foreignIndex],
+          };
+        }
+        return {
+          ...row,
+        };
+      });
+
+      return table(joinedTable);
+    },
+  };
+}
+
+export function createdBoxesTable(createdBoxes: CreatedBoxesResult[]) {
```

Comment:
atm only used in useCreatedBoxes hook, so I do not see a clear reason yet why to separate this out from the hook